src/peer_list.cpp

```diff
@@ -1043,13 +1037,11 @@ namespace libtorrent {
 		if (remote_address == address() || remote.port() == 0)
 			return nullptr;
 
-#if TORRENT_USE_IPV6
 		// don't allow link-local IPv6 addresses since they
 		// can't be used like normal addresses, they require an interface
 		// and will just cause connect() to fail with EINVAL
 		if (remote_address.is_v6() && remote_address.to_v6().is_link_local())
```

Comment:
Not really related to these changes, but link-local addresses could be allowed as long as `remote_address.to_v6().scope_id()` is non-zero.